qa/tasks/ceph_salt.py

```diff
@@ -0,0 +1,274 @@
+
+    def _install_ceph_salt(self):
+        '''
+        Installs ceph-salt on master either from source if repo and/or branch are 
+        provided in the suite yaml or from rpm if not
+        '''
+        global ceph_salt_ctx
+        if ceph_salt_ctx['repo']:
+            if not ceph_salt_ctx['branch']:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=ceph_salt_ctx['repo']
+                )
+            else:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=[ceph_salt_ctx['repo'], ceph_salt_ctx['branch']]
+                )
+        else:
+            self.scripts.run(
+            self.master_remote,
+            'install_ceph_salt.sh'
+            )
+        self.ctx.cluster.run(args=[ 'sudo', 'sh', '-c',
+                                    'systemctl restart salt-minion'])
+        self.master_remote.sh("sudo systemctl restart salt-master")
+
+            
+    def _populate_ceph_salt_context(self):
+        global ceph_salt_ctx
+        ceph_salt_ctx['log_anchor'] = self.config.get('log_anchor', self.log_anchor_str)
+        if not isinstance(ceph_salt_ctx['log_anchor'], str):
+            self.log.warning(
+                "log_anchor was set to non-string value ->{}<-, "
+                "changing to empty string"
+                .format(ceph_salt_ctx['log_anchor'])
+                )
+            ceph_salt_ctx['log_anchor'] = ''
+        ceph_salt_ctx['deploy'] = self.config.get('deploy', True)
+        ceph_salt_ctx['quiet_salt'] = self.config.get('quiet_salt', True)
+        ceph_salt_ctx['salt_manager_instance'] = SaltManager(self.ctx)
+        ceph_salt_ctx['master_remote'] = (
+                ceph_salt_ctx['salt_manager_instance'].master_remote
+                )
+        ceph_salt_ctx['repo'] = self.config.get('repo', None)
+        ceph_salt_ctx['branch'] = self.config.get('branch', None)
+
+    def _add_CA_repo(self):
+        '''
+        Adding `SUSE Internal CA Certificate` that's needed to install the container image
+        '''
+        self.ctx.cluster.run(args=[ 'sudo', 'sh', '-c', 
+                                    'sudo zypper -n addrepo --refresh --no-gpgcheck'
+                                    ' http://download.suse.de/ibs/SUSE:/CA/SLE_15_SP2/ '
+                                    '\'SUSE Internal CA Certificate\'; '
+                                    'zypper -n in ca-certificates-suse'])
+
+    def _rm_localhost_from_etc_hosts(self):
+        '''
+        Teuthology by default puts in '/etc/hosts' a line with the hostnames and localhost ip. This
+        causes salt to populate 'fqdn_ip4' with 127.0.0.1 and ceph-salt is breaking since it
+        has a exception for localhost resolution. So this function Removes localhost entries from
+        '/etc/hosts' in all hosts in order to use DNS server for names resolution.
+        '''
```

Comment:
I'm not sure this helps to make salt use DNS for hostname resolution, have checked a possibility to give ceph-salt predefined ip addresses?

Reply:
It's not available yet on `ceph-salt`. See https://github.com/ceph/ceph-salt/issues/64

Reply:
The following PR will allow the user to explicitly set bootstrap Mon IP address: https://github.com/ceph/ceph-salt/pull/156